File: justjavastaff/src/main/java/com/marknkamau/justjavastaff/data/network/OrdersRepository.java
package com.marknkamau.justjavastaff.data.network;

import com.marknkamau.justjavastaff.models.Order;
import com.marknkamau.justjavastaff.models.OrderItem;
import com.marknkamau.justjavastaff.models.OrderStatus;

import java.util.List;

/**
 * Created by Mark Njung'e.
 * mark.kamau@outlook.com
 * https://github.com/MarkNjunge
 */

public interface OrdersRepository {
    void getOrders(OrdersListener listener);

    void getOrderItems(String orderId, OrderItemsListener listener);

    void updateOderStatus(String orderId, OrderStatus status, BasicListener listener);

    interface BaseListener {
        void onError(String reason);
    }

    interface BasicListener extends BaseListener {
        void onSuccess();
    }

    interface OrdersListener extends BaseListener {
        void onSuccess(List<Order> orders);
    }

    interface OrderItemsListener extends BaseListener {
        void onSuccess(List<OrderItem> items);
    }
}

File: justjavastaff/src/main/java/com/marknkamau/justjavastaff/data/network/OrdersRepository.kt
package com.marknkamau.justjavastaff.data.network

import com.marknkamau.justjavastaff.models.Order
import com.marknkamau.justjavastaff.models.OrderItem
import com.marknkamau.justjavastaff.models.OrderStatus

/**
 * Created by Mark Njung'e.
 * mark.kamau@outlook.com
 * https://github.com/MarkNjunge
 */

interface OrdersRepository {
    fun getOrders(listener: OrdersListener)

    fun getOrderItems(orderId: String, listener: OrderItemsListener)

    fun updateOderStatus(orderId: String, status: OrderStatus, listener: BasicListener)

    interface BaseListener {
        fun onError(reason: String)
    }

    interface BasicListener : BaseListener {
        fun onSuccess()
    }

    interface OrdersListener : BaseListener {
        fun onSuccess(orders: List<Order>)
    }

    interface OrderItemsListener : BaseListener {
        fun onSuccess(items: List<OrderItem>)
    }
}
